src/behaviours/cluster/__snapshots__/order-of-sidebar-items.test.tsx.snap

```diff
@@ -0,0 +1,719 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```

Comment:
I think large snapshots are not a good thing. It makes it difficult to review and understand diffs, which can hide regressions, and in particular exposes rather technical details like classnames to the diff. I think snapshots, if used, should be kept as small as semantically reasonable. Alternatively, the snapshot assertion could be replaced with a manully defined assertion (about the order, for example for this test).

Reply:
I think large snapshots have more costs than benefits and I wouldn't approve this PR with these.

Reply:
I also don't like that snapshot based testing locks in the DOM, I think that the user doesn't care at all about it and more cares about the text then if an extra `<div>` is added somewhere.

Reply:
Again: Reasoning for snapshots is different.  Snapshots are very important for detecting unexpected changes in UI caused by some change in somewhere else.  

Please, provide more than you opinions. Concreate examples how you would solve the issue described here and in slack. Snapshots provide easy solution without minimal downsides (if any).

Reply:
I don't agree that this should be a requirement for testing. We want tests for UI components to test from a "user perspective", not asserting a specific DOM structure. If we have a good test coverage of unit tests of UI components we can be confident that the UI components work correctly, making snapshots unnecessary. I think that it's a rare edge case where these snapshots would be actionable for developers, thus making the downsides exceed the benefits.

The downsides of large snapshots are
a) difficult to review snapshot diffs, making them mostly useless
b) creating noise to diffs
c) having to update the snapshots when DOM changes
d) creating an incentive for developers to rely on a snapshot for asserting a semantic meaning of the DOM tree

Reply:
> If we have a good test coverage of unit tests

We don't have. And we won't have in near future. And we won't have ever, as long as whole team doesn't write code in TDD.

-- 
a) Large snapshot diffs are not supposed to be reviewed. That doesn't mean they are useless.
b) Read a)

c) It's not downside that you get instant feedback from unit tests that something has changed. Current situation in code base is that components are re-used in so many places that unexpected places need to be caught. And this doesn't affect only components, even though components are the ones that are rendered. How many bugs we have seen caused by some change in somewhere else? I can answer that: **too many**. If there is a way to give instant feedback to developer, we should definitely use it.

d) This is where PR review comes to place. No functionality should be tested with snapshots.

Reply:
Yes, the goal is to have good coverage.

The downside is having to do a manual step to update the snapshots (even if that step is pretty small).